src/git/repo.rs

```diff
@@ -965,6 +1077,134 @@ Either create it, or update the main branch setting by running:
             .map_err(wrap_git_error)?;
         Ok(make_non_zero_oid(oid))
     }
+
+    /// Amends the provided parent commit in memory and returns the resulting tree.
+    ///
+    /// Only amends the files provided in the options, and only supports amending from
+    /// either the working tree or the index, but not both.
+    ///
+    /// See `Repo::cherry_pick_fast` for motivation for performing the operation
+    /// in-memory.
+    pub fn amend_fast(
+        &self,
+        parent_commit: &Commit,
+        opts: &AmendFastOptions,
+    ) -> eyre::Result<Tree> {
+        let parent_commit_pathbufs = self
+            .get_paths_touched_by_commit(parent_commit)?
+            .ok_or_else(|| {
+                eyre::eyre!(
+                    "Could not get paths touched by commit: {:?}",
+                    &parent_commit
+                )
+            })?
+            .into_iter()
+            .collect_vec();
+        let changed_paths: Vec<PathBuf> = {
+            let mut result: HashSet<PathBuf> = parent_commit_pathbufs.into_iter().collect();
+            match opts {
+                AmendFastOptions::FromIndex { paths } => result.extend(paths.iter().cloned()),
+                AmendFastOptions::FromWorkingTree { ref status_entries } => {
+                    for entry in status_entries {
+                        result.extend(entry.paths().iter().cloned());
+                    }
+                }
+            };
+            result.into_iter().collect_vec()
+        };
+        let changed_paths = changed_paths
+            .iter()
+            .map(|path| path.as_path())
+            .collect_vec();
+
+        let dehydrated_parent =
+            self.dehydrate_commit(parent_commit, changed_paths.as_slice(), true)?;
+        let dehydrated_parent_tree = dehydrated_parent.get_tree()?;
+
+        let repo_path = self
+            .get_working_copy_path()
+            .expect("repo working copy path");
```

Comment:
Use `ok_or_else` here instead, so we can get a better stack trace.

Reply:
Done